node/mock_test.go

```diff
@@ -157,7 +165,8 @@ func (p *mockV0Provider) DeletePod(ctx context.Context, pod *v1.Pod) (err error)
 	}
 
 	p.notifier(pod)
-
+	// TODO (Sargun): Eventually delete the pod from the map. We cannot right now, because GetPodStatus can / will
```

Comment:
Maybe we can move it to a deleted list and delete it from the deleted list after the status has been collected from there?

Reply:
@cpuguy83 Given that this is changing with the new notifypods work shortly, I'm not inclined to do clever things here.

Reply:
Earlier, the approach I had was to look at deletionTimestamp, and do a timer.afterfunc and delete the pod (status) at deletiontimestamp time. That seems...complicated.